lib/oneview-sdk/resource/api200/logical_switch.rb

```diff
@@ -46,12 +46,13 @@ def create
         self
       end
 
-      # Updates this object using the data that exists on OneView
-      # @note Will overwrite any data that differs from OneView
-      # @return [Resource] self
-      def refresh
+      # Reclaims the top-of-rack switches in a logical switch.
+      # @return [OneviewSDK::LogicalSwitch] self
+      def refresh_state!
```

Comment:
Actually, the return is a task indicating wether the refresh is complete or not. If you want to return the Refreshed LogicalSwitch, you need to wait until the refresh state changes from Refreshing, then perform a retrieve!/refresh so this way you will have the refreshed LogicalSwitch.